packages/widgets/src/menubar.ts

```diff
@@ -382,6 +390,74 @@ export class MenuBar extends Widget {
     }
   }
 
+  /**
+   * A message handler invoked on an `'resize'` message.
+   */
+  protected _evtResize(event: Event): void {
+    let itemMenus = this.node.getElementsByClassName('lm-MenuBar-item');
+    let screenSize = this.node.offsetWidth;
+    let totalMenuSize = 0;
+    let index = -1;
+    let n = itemMenus.length;
+    let first = false;
+
+    if (this._menuItemSizes.length == 0) {
+      // Check if it is the first resize
+      first = true;
+    }
+
+    for (let i = 0; i < n; i++) {
+      let item = itemMenus[i] as HTMLLIElement;
+      totalMenuSize += item.offsetWidth;
+      if (first) {
+        // Add sizes to array
+        this._menuItemSizes.push(item.offsetWidth);
+      }
+      if (totalMenuSize > screenSize) {
+        index = i;
+        break;
+      }
+    }
+    if (first) {
+      first = false;
+    }
+    console.log(this._menuItemSizes);
+    if (index > -1) {
+      // Create hamburger menu
+      const commands = new CommandRegistry();
+      if (this._hamburgerMenu === null) {
+        this._hamburgerMenu = new Menu({ commands });
```

Comment:
You can solve the code scan warning using this:

```suggestion
        this._hamburgerMenu = new Menu({ commands: new CommandRegistry() });
```